std/path.d

```diff
@@ -3902,6 +3902,58 @@ if (isConvertibleToString!Range)
     assert(testAliasedString!isValidPath("/foo/bar"));
 }
 
+/** Checks whether `path1` is a subpath of `path2`
+  *
+  * Params:
+  *     path1 = string or Range of characters
+  *     path2 = string or Range of characters
+  *
+  * Returns:
+  *     true if `path` is a subpath of `path2`
+  *     false otherwise
+*/
+bool isSubPath(Range)(Range subPath, Range path)
+if ((isRandomAccessRange!Range && hasLength!Range && hasSlicing!Range &&
+    isSomeChar!(ElementEncodingType!Range) || isNarrowString!Range) &&
+    !isConvertibleToString!Range)
+{
+    import std.array : array;
+    
+    if (!isValidPath(subPath) || !isValidPath(path)) return false;
+
+    auto subPathAbsolute = asAbsolutePath(subPath).array;
+    auto pathAbsolute = asAbsolutePath(path).array;
+    subPathAbsolute = buildNormalizedPath(subPathAbsolute);
+    pathAbsolute = buildNormalizedPath(pathAbsolute);
+
+    auto subPathComponents = pathSplitter(subPathAbsolute).array;
+    auto pathComponents = pathSplitter(pathAbsolute).array;
+
+    if (subPathComponents.length > pathComponents.length) return false;
+
+    for (uint i = 0; i < subPathComponents.length; i++)
+        if (!(subPathComponents[i] == pathComponents[i])) return false;   
+
+    return true;
+}
+
+@safe unittest
+{
+    version (Posix)
+    {
+        assert(isSubPath("/foo/bar","/foo/bar/some"));
+        assert(!isSubPath("/foo/bar/some/file/..", "/foo/bar"));
+        assert(!isSubPath("/foo/bar", "/foo/bars"));
+    }
+
+    version (Windows)
+    {
+        assert(isSubPath("c:/foo/", "c:/foo/bar"));
+        assert(isSubPath("c:/foo/..", "c:/foo/bar/../."));
+        assert(!isSubPath("c:/foo", "c:/foobar"));
+    }
+}
```

Comment:
- Please add tests where the paths are a range.
- Please add tests where both paths are relative.
- Please add tests where only one of the paths is relative, and the result depends on the current directory (assuming that we decide to keep `asAbsolutePath`).